### PR TITLE
Add the `--unroll` option

### LINKED ISSUE #1928: Add an option to specify the recursion unrolling depth
An option e.g. `--unroll N` would globally override the default depth 140.

Later, we should have a mechanism to specify unrolling depth on a per-function basis.

### PR DESCRIPTION
**Pull request by lukaszcz**
* Closes #1928

### CODE REVIEW

File: src/Juvix/Compiler/Pipeline/EntryPoint.hs [hunk 1]
```diff
@@ -44,8 +45,9 @@ defaultEntryPoint root mainFile =
       _entryPointStdin = Nothing,
       _entryPointPackage = defaultPackage root buildDir,
       _entryPointGenericOptions = defaultGenericOptions,
-      _entryPointTarget = TargetCNative64,
```
Comment (janmasrovira) [inline src/Juvix/Compiler/Pipeline/EntryPoint.hs:47]: was this change intended?

Comment (lukaszcz) [reply src/Juvix/Compiler/Pipeline/EntryPoint.hs:47]: It shouldn't matter, because if you need a different target you should override it. I tried to make defaults consistent.

**Review by janmasrovira — APPROVED** [review]
